### PR TITLE
Support for multiple Ceph RBD version's output format

### PR DESCRIPTION
**Pull request by osnyx**
Follow-up to #23.

This makes backy compatible to the Ceph Nautilus client tooling: rbd showmapped has now a different json output. It now returns a list of objects (dicts) decribing mappings, instead of an object (dictionary) already at the top level.
Keeps compatibility to previous Ceph versions by checking the Ceph version at runtime and differing in behaviour upon that.

Unit tests check behaviour based on outputs from Ceph Jewel, Luminous, and Nautilus.

Manually tested with #26 and #27 already merged.
Formatting issues have been (partially) neglected, as changes are going to be enforced by black through pre-commit rules anyways after merging #22.

## Release process

Impact: None, just wider compatibility.

Changelog:
- add compatibility to RBD backends running on Ceph Nautilus
- keep compatibility to RBD backends running on Ceph Jewel or Luminous

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined?
  - [x] behaviour changes acommodating changed input formats must continue to support existing (legacy) systems
  - [x] test coverage of system must not decrease after change
  - [x] validation agains real-world data
- [x] Security requirements tested? (EVIDENCE)
  - [x] unit tests covering all supported Ceph versions and their input formats
  - [x] test data acquired from real-world systems
  - [x] test Nautilus compitability in development cluster
  - note: compatibility against Ceph Jewel and Luminous will **not** be tested manually due to the lack of a development clusters. The code covering these systems remains almost unchanged though and is covered by unit tests.

### CODE REVIEW

**Comment by osnyx** [comment]
All tests are fixed now, ready for review.

One possible blocker: I see 1 backup failure in dev, verly likely due to corrupt parent revision data. We need to look into this before merging. PL-131339

**Comment by osnyx** [comment]
@dhnasa and I think that the mentioned backup failure is not a new regression, but an extended case of #17 and the old revisions not having been distrusted despite failing their verifications in older backy versions.

File: src/backy/sources/ceph/rbd.py [hunk 1]
```diff
@@ -13,39 +13,85 @@
...
     def map(self, image, readonly=False):
-        self._rbd(["--read-only" if readonly else "", "map", image])
-
-        mappings = self._rbd(["showmapped"], format="json")
-        for mapping in mappings.values():
-            if image == "{pool}/{name}@{snap}".format(**mapping):
-                return mapping
-        raise RuntimeError("Map not found in mapping list.")
+        def parse_mappings_legacy(mappings):
+            """The parser code for Ceph release Luminous and earlier."""
```
Comment (ctheune) [inline src/backy/sources/ceph/rbd.py:53]: parse_mappings_pre_luminous

File: src/backy/sources/ceph/rbd.py [hunk 2]
```diff
@@ -13,39 +13,85 @@
...
+            for mapping in mappings.values():
+                if image == "{pool}/{name}@{snap}".format(**mapping):
+                    return mapping
+            raise RuntimeError("Map not found in mapping list.")
+
+        def parse_mappings_current(mappings):
```
Comment (ctheune) [inline src/backy/sources/ceph/rbd.py:59]: parse_mappings_since_luminous + comment is copy/pasta :)

**Review by ctheune — CHANGES_REQUESTED** [review]
looks good in general, really few small changes and we're good to go

**Review by ctheune — APPROVED** [review]
lgtm

**Pull request merged**